autogluon/utils/tabular/features/generators/abstract.py

```diff
@@ -0,0 +1,520 @@
+
+        Parameters
+        ----------
+        features : list of str
+            List of feature names to remove from the expected input.
+        """
+        if features:
+            self.feature_metadata_in = self.feature_metadata_in.remove_features(features=features)
+            self.features_in = self.feature_metadata_in.get_features()
+
+    def _rename_features_in(self, column_rename_map: dict):
+        if self.feature_metadata_in is not None:
+            self.feature_metadata_in = self.feature_metadata_in.rename_features(column_rename_map)
+        if self.features_in is not None:
+            self.features_in = [column_rename_map.get(col, col) for col in self.features_in]
+
+    def _rename_features_out(self, column_rename_map: dict):
+        self.feature_metadata = self.feature_metadata.rename_features(column_rename_map)
+        self.feature_metadata_real = self.feature_metadata_real.rename_features(column_rename_map)
+        self.features_out = [column_rename_map.get(col, col) for col in self.features_out]
+
+    def _get_renamed_features(self, X: DataFrame) -> (DataFrame, dict):
+        X_columns_orig = list(X.columns)
+        X_columns_new = list(X.columns)
+        if self._name_prefix:
+            X_columns_new = [self._name_prefix + column for column in X_columns_new]
+        if self._name_suffix:
+            X_columns_new = [column + self._name_suffix for column in X_columns_new]
+        if X_columns_orig != X_columns_new:
+            is_updated_name = True
+        else:
+            is_updated_name = False
+        column_rename_map = {orig: new for orig, new in zip(X_columns_orig, X_columns_new)}
+        return column_rename_map, is_updated_name
+
+    def _post_fit_cleanup(self):
+        """
+        Any cleanup operations after all metadata objects have been constructed, but prior to feature renaming, should be done here
+        This includes removing keys from internal lists and dictionaries of features which have been removed, and deletion of any temp variables.
+        """
+        pass
+
+    # TODO: Move to a generator
+    @staticmethod
+    def _get_useless_features(X: DataFrame) -> list:
+        useless_features = []
+        for column in X:
+            if is_useless_feature(X[column]):
+                useless_features.append(column)
+        return useless_features
+
+    def set_log_prefix(self, log_prefix, prepend=False):
+        if prepend:
+            self.log_prefix = log_prefix + self.log_prefix
+        else:
+            self.log_prefix = log_prefix
+
+    def set_verbosity(self, verbosity: int):
+        self.verbosity = verbosity
+
+    def _log(self, level, msg, log_prefix=None, verb_min=None):
+        if self.verbosity == 0:
+            return
+        if verb_min is None or self.verbosity >= verb_min:
+            if log_prefix is None:
+                log_prefix = self.log_prefix
+            logger.log(level, f'{log_prefix}{msg}')
+
```

Comment:
This functionality looks like very generic one, can we move it into a separate mixin class?

Reply:
Added TODO